requirements.txt

```diff
@@ -1,8 +1,13 @@
+backoff
+datashader
 fiona
 geopandas
 h5py
+holoviews
```

Comment:
is geoviews a dependency of holoviews? If so, we'll need to do the same gymnastics we did before for those who haven't/can't install geoviews. Relatedly (though not to these particular lines of code), we should add a few checks/messages for users trying to use/import the visualization module who don't have geoviews. Otherwise they'll get top level import errors (we might be able to simplify this by just doing it once in the `__init__.py`, though it might make sense to have it explicitly in the places it will become an issue).

Reply:
Looks like `[holoviews](https://github.com/holoviz/holoviews/blob/master/setup.py)` doesn't require `cartopy` or `geoviews`, I removed `geoviews` from `Visualize` as it's only required to plot spatial extent, which was moved to `Query` and `geoviews` dependency has been solved, just tested in a fresh environment using `pip install -e.` and everything is working fine.

Reply:
Yeah it's a good idea to add checks for `geoviews import` error, should we just check this import error then tell them different ways of installing `geoviews` depending on whether they `conda install icepyx` or `pip install icepyx`?

Reply:
Awesome! At some point I think we discussed moving `Query.visualize_spatial_extent()` to the `Visualize` module (obviously, in either case we'll keep it so that `geoviews` is only an optional dependency and this method can be called directly from the Query object). Thoughts? I haven't fully argued this out with myself yet.

Reply:
Just watched the webinar, great talk! Yeah we discussed it before about migrating `visualize_spatial_extent()` from https://github.com/icesat2py/icepyx/pull/176 back to the `Visualize` module, but I am not sure if it's necessary now...since everything all ended up in `Query` module, we could just call `Query.visualize_spatial_extent()` for spatial extent and `region.visualize_elevation()` for elevation visualization directly.

Reply:
Agree with @icetianli to keep `visualize_elevation()` where it is now in the `Query` object. I'm on the fence on whether to make `holoviews` a required dependency of `icepyx` because it has a lot of sub-dependencies (though not as bad as geoviews), but it seems inevitable if we're moving into exploratory data analysis work (people like fancy GUIs) so I'm ok with adding it.

Reply:
Sounds good on adding holoviews as a dependency. To clarify, either way we'd have `visualize_elevation()` and `visualize_spatial_extent()` methods for Query objects; the bigger question is where the implementation of those lives (in the query or visualization modules). My thinking was that this is analogous to the other modules (e.g. granules), which very early on were part of a giant query module and ultimately separated out into submodules with their own class objects to break the code into more reasonably sized units. If, as @weiji14 notes, we get more into including GUIs for data analysis work, then I suspect having all of the visualization components stored in one `Visualization` object is going to be smoother. I will work on some of these changes as I was envisioning them and we can just revert the commit if it makes things worse.

Reply:
Sounds like a plan!

Reply:
I just pushed my version of this, including examples of each access method at the end of your example notebook. This approach has the added benefit of using the checks built into `Query` to validate the user's inputs.